### PR TITLE
Imohash implementation

### PR DESCRIPTION
**Pull request by oertl**

### CODE REVIEW

File: src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+
+class Imohash1_0_2 extends AbstractFileHasher128 {
+
+  static final long DEFAULT_SAMPLE_THRESHOLD = 128 * 1024;
+  static final int DEFAULT_SAMPLE_SIZE = 16 * 1024;
+
+  private static final int BUFFER_SIZE = 1024;
```
Comment (mrgreywater) [inline src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java:59]: I'd increase the buffer size to at least 4KB, as thats the common default block size for ntfs and ext4.

File: src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  static FileHasher128 create(int sampleSize, long sampleThreshold) {
+    return new Imohash1_0_2(sampleSize, sampleThreshold);
+  }
+
+  private void processBytes(long numBytes, InputStream stream, HashStream128 hashStream)
+      throws IOException {
+    long numBytesRemaining = numBytes;
+    int bufferPos = 0;
+    while (numBytesRemaining > 0) {
+      int numBytesRequested = buffer.length - bufferPos;
```
Comment (mrgreywater) [inline src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java:91]: I feel like the names numBytesRemaining and numBytesRequested are swapped.

File: src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java (hunk 1 above)
Comment (oertl) [reply src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java]: changed buffer size to 4kB

File: src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java (hunk 2 above)
Comment (oertl) [reply src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java]: renamed variable for better clarity

**Pull request merged**